### DISCUSSION TITLE
Create sensor combining conditions of 2 dp

### DISCUSSION THREAD

**Discussion by JackJackattack4** [make-all/tuya-local#1335] [Q&A] [answered]
I have a specific situation to resolve which I did not see elsewhere.

My Tuya device uses a strange way to share information. It uses one dp to select the parameters that can be inquired and uses a different dp to provide the numerical value of that parameter (output).

For example: When DP ID 104 is 5 (inlet water temp.), the output of dp ID 105 (an integer) will return the numerical value of the corresponding parameter (inlet water temp.).

The ideal case, the information of each parameter would be extracted and stored as sensors in HA. The code below is as far I could reach so far (not much).

Maybe someone has had the same situation to solve?

See code below:
...
secondary_entities:
  - entity: select
    name: Inquire_ID
    category: config
    dps:
      - id: 104
        # mappings based on dev portal info, observed values and manufacturer's manual. DPID 104 (para. Inquiry) specifies which parameter is inquired and the value is returned using DPID 105 (Inquiry Value).
        type: string
        name: option
        mapping:
          - dps_val: "1"
            value: version no.
          - dps_val: "2"
            value: ambient temp.
          - dps_val: "3"
            value: water tank temp.
          - dps_val: "4"
            value: outlet water temp.
          - dps_val: "5"
            value: inlet water temp.
          - dps_val: "6"
            value: evaporator coil temp.
          - dps_val: "7"
            value: exhaust gas temp.
          - dps_val: "8"
            value: condenser coil temp. (after throttling)
          - dps_val: "9"
            value: suction temp.
          - dps_val: "10"
            value: module temp.
          - dps_val: "11"
            value: main valve opening degrees
          - dps_val: "12"
            value: auxiliary valve opening degrees
          - dps_val: "13"
            value: exhaust gas valve opening degrees
          - dps_val: "14"
            value: Fan speed
          - dps_val: "15"
            value: running frequency
          - dps_val: "16"
            value: AC voltage
          - dps_val: "17"
            value: DC voltage
          - dps_val: "18"
            value: current of whole unit
          - dps_val: "19"
            value: compressor current
          - dps_val: "20"
            value: output power
          - dps_val: "21"
            value: EVI plate heat exchanger inlet sensor
          - dps_val: "22"
            value: EVI plate heat exchanger outlet sensor
          - dps_val: "23"
            value: low pressure
          - dps_val: "24"
            value: high pressure
          - dps_val: "25"
            value: error code
          - dps_val: "26"
            value: limited frequency code
          - dps_val: "27"
            value: LCD Controller version
          - dps_val: "28"
            value: MCU1 version
          - dps_val: "29"
            value: MCU2 version
          - dps_val: "30"
            value: MCU3 version
          - dps_val: "31"
            value: Error History
  #- entity: select
    #name: Inquire_Value
    #category: config
    #dps:
      #- id: 105
        # DPID 104 (para. Inquiry) specifies which parameter is inquired and the value is returned using DPID 105 (Inquiry Value).
        #type: integer
        #name: option
  - entity: select
    name: Set_ID
    category: config
    dps:
      - id: 106
        # mappings based on dev portal info, observed values and manufacturer's manual. DPID 106 (Para. Setting) specifies which parameter is set using the value in DPID 107 (Set Value).
        type: string
        name: option
        mapping:
          - dps_val: "d01"
            value: Heating Temp Setting
          - dps_val: "d02"
            value: Heating Water Tank Temp Setting
          - dps_val: "d03"
            value: Cooling Temp Setting
          - dps_val: "d04"
            value: Restart Temp Difference 
          - dps_val: "d05"
            value: Selection on Control basis
          - dps_val: "d06"
            value: Setting Temp Of Electric Heater
          - dps_val: "d07"
            value: Deviation Time of Electric Heater Start
          - dps_val: "d08"
            value: Evaporator coil Temp to enter into defrosting
          - dps_val: "d09"
            value: Evaporator coil Temp to exit defrosting 
          - dps_val: "d10"
            value: Defrosting Cycle Period
          - dps_val: "d11"
            value: Defrosting Time
          - dps_val: "d12"
            value: Selection on Control after temp is reached
          - dps_val: "d13"
            value: Ambient temp for Stopping and Starting machine
          - dps_val: "d14"
            value: Temp Difference of Inlet and Outlet water for heat pump's
          - dps_val: "d15"
            value: Working Way of Water Pump
  #- entity: select
    #name: Set_Value
    #category: config
    #dps:
      #- id: 107
        # DPID 106 (Para. Setting) specifies which parameter is set using the value in DPID 107 (Set Value).
        #type: integer
        #name: option

**Comment by make-all** [comment, marked as answer]
Combining different values into a single one seems like a job for template sensors, not for this integration.  I would just return the values from the dps as extra attributes on the main entity, and do your processing into a sensor using the template sensor  integration.

**Comment by JackJackattack4** [comment]
Understood. I will be creating these sensors outside of this integration. I updated the script. It extracts all dps available now. Thanks.
